### DISCUSSION TITLE
Moving inventree data/installation from windows to linux/docker

### DISCUSSION THREAD

**Discussion by t0roide** [inventree/InvenTree#3149] [Q&A] [answered]
Hi,

I've been using actively inventree at work  for a couple weeks now and loving it so far, the initial installation i.e. the one we're currently using is running on docker on my windows laptop but since my computer is under heavy use and not always available i want to migrate everything to a pc running docker on linux that will host the inventory and other serivices. I read on the docs that migration of the database is possible by exporting a .json file and then importing it back on the new installation but i would like to know if this means than only data about the parts can be transferred, what about the product images? supplier/manufacturer information and image? and what about the users and configurations like that.

From my laptop i've loaded some company logos and also some parts images and i know that those are stored in my persistent data folder because i can see them on there, but i'm not sure about images that other users loaded because those i can't see locally. Is there a way to migrate as much data/images/configurations as possible? I can think of a combination of copying the persistent data folder + .json importing but since the persistent data folder has only the things i've added i want to be sure of how this could work or what are my options to make the migration as smooth as possible without data loss.

Thanks in advance!

**Comment by matmair** [comment, marked as answer]
Hi there @isa-tt !
The commands [described here](https://inventree.readthedocs.io/en/latest/start/migrate/) export everything in the database:. You then just need to copy the resulting file, your persistent folder and your secrets.txt and config.yaml to the new host. Import the data and you should be up and running again.

**Reply by t0roide** [reply]
Great! but where i'm supposed to find the secrets.txt file? i don't think i've seen it when snooping around the folders

**Reply by matmair** [reply]
I can not tell you that as that is variable - see [the docs](https://inventree.readthedocs.io/en/latest/extend/api/) for details.

**Reply by t0roide** [reply]
i'll give it a try without the secrets.txt file because i couldn't find it anywhere within my docker/inventree related files.

**Reply by SchrodingersGat** [reply]
@matmair I do  not believe the standard setup includes a `secrets.txt` file - there is `secret_key.txt` however

**Reply by t0roide** [reply]
yes! @SchrodingersGat the secret_key.txt file is inside the persistent data folder, as well as a data.json file that seems to be up to date according to the modification date. I tried to migrate to docker on linux mint but i can't get it to work, when exporting the .json file even if i choose the location it saves it to /home/inventree on my windows machine, this location doesn't exist (don't know where it is), or does it exist only within docker? if that's the case how do i access it?. So far i've been thinking that the .json file on the persistent data folder is a valid file to import the data but when i do so using the command
`docker compose run inventree-server invoke import-records -f data.json`

i get the following error,
`Error: File '/home/inventree/data.json' does not exist`

Sorry if this questions are basic but i'm a newbie at docker and even reading the docs sometimes it's still hard to get what's going on in the background (Maybe this questions are too docker focused, sorry about that). And thanks for the help so far!

**Reply by t0roide** [reply]
What i've tried so far is copying the persistent data folder to the new machine, copying the .env and docker-compose.yml files to the inventree folder on the new machine home directory. From there if i follow the installation directions from the docs, i run this on the production folder
`docker-compose run inventree-server invoke update`
then i run the migration command as 
`docker compose run inventree-server invoke import-records -f data.json`
and the error is the one i wrote above, because i don't know where to put the data.json file or where for sure it is, i know that it is in my persistent folder but the command can't reach for some reason i'm not understanding.
Thought this could be helpful info to get help!

**Reply by matmair** [reply]
@isa-tt Sounds like either a permission error or you are looking in the wrong place.
Please try using a full path instead of a relative file name that could point to a lot of different places depending on where your console is.
It might also help if you read up on [storage in Docker](https://docs.docker.com/storage/). Where your persistent folder is mounted or not really depends on how you set up your environment - it is difficult to make general assumptions about that.

**Reply by t0roide** [reply]
Thanks @matmair for the link to the docker info, i've been reading some more on the whole docker volumes for windows thing and what has worked for me is this 
https://stackoverflow.com/questions/43181654/locating-data-volumes-in-docker-desktop-windows
i'll leave the link just in case someone is trying to figure this out too, following the info on that link the wsl folder for docker appeared in my network connections and i could look there for the target directory /home/inventree that was listed on the volumes in the docker desktop app. After looking around i just used the search bar to look for an inventree folder or .json file from the last export and found everything.

So no more missing data.json file to migrate my data and found the home directory for my container, but i have another question, before trying to migrate, the persistent data folder needs some update or does it updates itself while docker is running? if it needs updating which command should i use?

Some other info on the docker volumes that helped and was newbie friendly:

- https://earthly.dev/blog/docker-volumes/#:~:text=Unlike%20bind%20mount%2C%20where%20you,%2C%20restore%2C%20and%20migration). 

- https://www.freecodecamp.org/news/where-are-docker-images-stored-docker-container-paths-explained/

**Reply by matmair** [reply]
@isa-tt Great!
The data is kept in the persistent folder - you just need to copy that.

**Reply by matmair** [reply]
@t0roide can we mark this as answered or do you need more info?

**Reply by t0roide** [reply]
@matmair yes! consider this answered, all the info above was enough for me to solve my questions.

**Reply by matmair** [reply]
Feel free to hit us up if you encounter any further issues
